### PR TITLE
add an "unstable features" chapter to the rustdoc book

### PR DESCRIPTION
**Pull request by QuietMisdreavus**
There are several rustdoc features that currently are undocumented, but also don't fit with the rest of the Rustdoc Book since they're also unstable. Some of these have corresponding feature gates and chapters in the Unstable Book, but many don't, and i wanted a place to talk about them officially.

Goal: talk about everything rustdoc can do that needs nightly

- [x] Feature gates (extensions to the doc attribute that can be caught by the compiler)
  - [x] doc(cfg)
  - [x] doc(masked)
  - [x] doc(spotlight)
  - [x] doc(include)
- [x] Command-line flags (features that require a CLI flag to use, where the flag itself is a `-Z` command or otherwise requires `-Z unstable-options` before rustdoc will accept it)
  - [x] markdown-before-content/markdown-after-content
  - [x] playground-url
  - [x] display-warnings
  - [x] crate-version
  - [x] linker
  - [x] sort-modules-by-appearance
  - [x] themes/theme-checker
  - [x] resource-suffix
  - [x] `-Z force-unstable-if-unmarked`
- [x] Nightly-gated functionality (features that are gated by requiring a nightly build without needing a CLI flag or a feature gate to unlock)
  - [x] intra-links
  - [x] error numbers for `compile_fail` doctests

### CODE REVIEW

**Comment by QuietMisdreavus** [comment]
While investigating `--display-warnings`, i had trouble coming up with a good explanation of what it did that either didn't have a bug or wasn't possible with other flags. I wrote up my investigation in the "tracking issue" for it: https://github.com/rust-lang/rust/issues/41574#issuecomment-373758210

**Comment by QuietMisdreavus** [comment]
I'm not sure whether to document `-Z force-unstable-if-unmarked, since it's purely an implementation detail of the compiler and std. (I'm also not that certain what it does, other than "pass the same flag to `rustc`", but that might be explanation enough, in lieu of a proper explanation somewhere else. `>_>`)

Otherwise, short of a decision on `--display-warnings`, this is ready to go!

@rust-lang/docs Can y'all take a look at this and make sure i'm not being misleading or specifying too much?

**Comment by QuietMisdreavus** [comment]
I've gone ahead and added sections about `--display-warnings` and `-Z force-unstable-if-unmarked`. As far as i know, this chapter is complete; please take a look!

File: src/doc/rustdoc/src/unstable-features.md [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+when a given type implements one of these traits and call special attention to it when a function
+returns one of these types. This is the "Important Traits" dialog, visible as a circle-i button next
+to the function, which, when clicked, shows the dialog.
+
+In the standard library, the traits that qualify for inclusion are `Iterator`, `io::Read`, and
+`io::Write`. However, rather than being implemented as a hard-coded list, these traits have a
+special marker attribute on them: `#[doc(spotlight)]`. This means that you could apply this
+attribute to your own trait to include it in the "Important Traits" dialog in documentation.
+
+The `#[doc(spotlight)]` attribute is controlled by a feature gate. For more information, see [its
```
Comment (frewsxcv) [inline src/doc/rustdoc/src/unstable-features.md]: > is controlled by a feature gate

would it be good to mention the name of the feature gate directly in this document?

File: src/doc/rustdoc/src/unstable-features.md [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+
+As designed in [RFC 1990], Rustdoc can read an external file to use as a type's documentation. This
+is useful if certain documentation is so long that it would break the flow of reading the source.
+Instead of writing it all inline, writing `#[doc(include = "sometype.md")]` (where `sometype.md` is
+a file adjacent to the `lib.rs` for the crate) will ask Rustdoc to instead read that file and use it
+as if it were written inline.
+
+[RFC 1990]: https://github.com/rust-lang/rfcs/pull/1990
+
+`#[doc(include = "...")]` is currently controlled by a feature gate. For more information, see [its
```
Comment (frewsxcv) [inline src/doc/rustdoc/src/unstable-features.md]: > is currently controlled by a feature gate

would it be good to mention the name of the feature gate directly in this document?

**Review by frewsxcv — APPROVED** [review]
this is soo great, awesome work @QuietMisdreavus 🎉

File: src/doc/rustdoc/src/unstable-features.md (hunk 1 above)
Comment (QuietMisdreavus) [reply src/doc/rustdoc/src/unstable-features.md]: I think my original idea was that by linking to the Unstable Book, people could easily find the name of the feature gate that way, but it's not that much more to write it out here as well. I'll add it.

**Comment by QuietMisdreavus** [comment]
@frewsxcv What do you think of it now?

**Review by frewsxcv — APPROVED** [review]

**Comment by frewsxcv** [comment]
r=me if you don't have any other additions

**Comment by QuietMisdreavus** [comment]
@bors r=frewsxcv

Nope, that's all i had! Thanks so much!

**Pull request merged**